internal/configs/virtualserver_test.go

```diff
@@ -894,12 +929,109 @@ func TestGenerateUpstreamForZeroEndpoints(t *testing.T) {
 		Servers: nil,
 	}
 
-	result = generateUpstream(name, upstream, endpoints, isPlus, &cfgParams)
+	result = generateUpstream(name, upstream, vsEx, isPlus, isResolvedConfigured, &cfgParams)
 	if !reflect.DeepEqual(result, expectedForNGINXPlus) {
 		t.Errorf("generateUpstream(isPlus=%v) returned %v but expected %v", isPlus, result, expectedForNGINXPlus)
 	}
 }
 
+func TestGenerateUpstreamForExternalNameService(t *testing.T) {
+	name := "test-upstream"
+	vsEx := &VirtualServerEx{
+		VirtualServer: &conf_v1alpha1.VirtualServer{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:      name,
+				Namespace: "test-namespace",
+			},
+		},
+		Endpoints: map[string][]string{
+			"test-namespace/test-upstream:0": []string{"example.com"},
+		},
+		ExternalNameSvcs: map[string]bool{
+			name: true,
+		},
+	}
+	cfgParams := ConfigParams{
+		MaxFails:    1,
+		FailTimeout: "10s",
+	}
+
+	tests := []struct {
+		upstream             conf_v1alpha1.Upstream
+		vsEx                 *VirtualServerEx
+		isPlus               bool
+		isResolverConfigured bool
+		expected             version2.Upstream
+	}{
+		{
+			upstream:             conf_v1alpha1.Upstream{},
+			vsEx:                 vsEx,
+			isPlus:               false,
```

Comment:
for NGINX OSS, ExternalNameSvcs map will always be empty, as mandated by the following code https://github.com/nginxinc/kubernetes-ingress/blob/211b8f98bb2cbf4e1e83be040b161ff58707465d/internal/k8s/controller.go#L1521 This is a precondition for the `generateUpstream` function and if this precondition is not true -- it is a bug in the code, not invalid input. That's why I don't think `isPlus: false` cases are necessary.  To avoid confusion, perhaps it makes sense to add a comment in the `generateUpstream` function that `isExternalNameSvc` is never true for OSS?

Reply:
The tests is important to check that we don't fill `ExternalNameSvcs` map in OSS. If we change the way we fill that by mistake, this test should fail and we will catch it (resolve=True for OSS fails) no?